src/platform/src/lib.rs

```diff
@@ -54,6 +61,10 @@ pub unsafe fn c_str_n(s: *const c_char, n: usize) -> &'static [u8] {
     slice::from_raw_parts(s as *const u8, size as usize)
 }
 
+pub unsafe fn cstr_from_bytes_with_nul_unchecked(bytes: &[u8]) -> *const c_char {
+    &*(bytes as *const [u8] as *const c_char)
```

Comment:
The `&*` is only needed in CStr because that function returns a reference, not a pointer.

And it looks like the use of a pointer to a slice is because of the non-optimal way CStr is defined: 

```rust
pub struct CStr {
    // FIXME: this should not be represented with a DST slice but rather with
    //        just a raw `c_char` along with some form of marker to make
    //        this an unsized type. Essentially `sizeof(&CStr)` should be the
    //        same as `sizeof(&c_char)` but `CStr` should be an unsized type.
    inner: [c_char]
}
```

Using the cast `as *const [u8] as *const c_char` may work, but I think that's potentially undefined behavior. `as_ptr()` exists for this purpose.